src/ui/ToggleEditor.tsx

```diff
@@ -357,7 +341,7 @@ class ToggleEditor extends Component<ToggleEditorProps, ToggleEditorState> {
           See the JS console for more detailed reporting.`;
         }
         this.copyMarks(oldAst);                         // Preserve old TextMarkers
-        this.currentCode = code;                        // update CM with the PP code
+        this.state.code = code;                         // update CM with the PP code
```

Comment:
You shouldn't assign to state like this outside of the constructor. You're already inside a call to `setState()`, so why not just pass the new code value through like you do with blockMode a few lines below?

```js
return {...state, code: code, blockMode: blockMode};
```

Reply:
Done